infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesClientFactory.java

```diff
@@ -220,4 +236,44 @@ public UnclosableKubernetesClient(OkHttpClient httpClient, Config config) {
     @Override
     public void close() {}
   }
+
+  /**
+   * Counts client invocations. Holds the value in it's own {@link AtomicInteger} counter.
+   *
+   * <p>In case that metrics are disabled, this class can be constructed in Noop mode with
+   * `ClientInvocationCounter(false)`.
+   */
+  private static class ClientInvocationCounter implements Runnable {
```

Comment:
What about making this an interface and having 2 impls in place of this 1 where 1 specialization would count and the other would just return 0 in `getCount` and do nothing in `run()`. Just to avoid the if statement on every invocation of which there are going to be many.